File: app/src/main/java/com/mikelau/sample/utils/view/CardPaddingItemDecoration.java
package com.mikelau.sample.utils.view;


import android.content.Context;
import android.graphics.Rect;
import android.util.TypedValue;
import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

public class CardPaddingItemDecoration extends RecyclerView.ItemDecoration {

    private int paddingBetweenItems;

    public CardPaddingItemDecoration(Context context) {
        this.paddingBetweenItems = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 8, context.getResources().getDisplayMetrics());
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
        outRect.set(0, 0, 0, paddingBetweenItems);
    }
}

File: app/src/main/java/com/mikelau/shimmerxsample/utils/CardPaddingItemDecoration.kt
package com.mikelau.shimmerxsample.utils

import android.content.Context
import android.graphics.Rect
import android.util.TypedValue
import android.view.View
import androidx.recyclerview.widget.RecyclerView

class CardPaddingItemDecoration(context: Context) : RecyclerView.ItemDecoration() {

    private val paddingBetweenItems: Int

    init {
        paddingBetweenItems = TypedValue.applyDimension(
            TypedValue.COMPLEX_UNIT_DIP,
            8f,
            context.resources.displayMetrics
        ).toInt()
    }

    override fun getItemOffsets(
        outRect: Rect,
        view: View,
        parent: RecyclerView,
        state: RecyclerView.State
    ) {
        outRect[0, 0, 0] = paddingBetweenItems
    }
}
